### PR TITLE
Fix WasbHook.delete_file broken when using prefix

### LINKED ISSUE #15636: WasbHook does not delete blobs with slash characters in prefix mode
**Apache Airflow version**: 2.0.1

**Environment**:

- **Cloud provider or hardware configuration**: docker container
- **OS** (e.g. from /etc/os-release): `Debian GNU/Linux 10 (buster)`
- **Kernel** (e.g. `uname -a`): `Linux 69a18af222ff 3.10.0-1160.15.2.el7.x86_64 #1 SMP Thu Jan 21 16:15:07 EST 2021 x86_64 GNU/Linux`
- **Install tools**: `pip`
- **Others**: `azure` extras

**What happened**:

`airflow.providers.microsoft.azure.hooks.wasb.WasbHook.delete_file` unable to delete blobs if both conditions below are true:

* `is_prefix` argument set to `True`
* the target blobs contain at least one '/' character in their names

**What you expected to happen**:
All files starting with the specified prefix are removed.

The problem is caused by this line: https://github.com/apache/airflow/blob/73187871703bce22783a42db3d3cec9045ee1de2/airflow/providers/microsoft/azure/hooks/wasb.py#L410

By not specifying `delimiter = ''` the listed blobs won't be terminal blobs if the target blobs happen to contain the default delimiter (`/`) in their name

**How to reproduce it**:

For example, consider the following scenario. We have a blob container `cnt` with two files:
* `parent/file1`
* `parent/file2`

The following call should delete both files:
``` python
wasb_hook= WasbHook(...)
wasb_hook.delete_file(
    container_name='cnt',
    blob_name='parent',
    is_prefix=True,
)
```
But instead we get an error `Blob(s) not found: ('parent/',)`

### PR DESCRIPTION
**Pull request by monti-python**
closes: #15636

### CODE REVIEW

File: airflow/providers/microsoft/azure/hooks/wasb.py [hunk 1]
```diff
@@ -407,7 +407,7 @@ def delete_file(
         :type kwargs: object
         """
         if is_prefix:
-            blobs_to_delete = self.get_blobs_list(container_name, prefix=blob_name, **kwargs)
+            blobs_to_delete = self.get_blobs_list(container_name, prefix=blob_name, delimiter='', **kwargs)
```
Comment (ephraimbuddy) [inline airflow/providers/microsoft/azure/hooks/wasb.py]: I prefer this delimiter is not hardcoded.

**Review by ephraimbuddy — CHANGES_REQUESTED** [review]

File: airflow/providers/microsoft/azure/hooks/wasb.py (hunk 1 above)
Comment (monti-python) [reply airflow/providers/microsoft/azure/hooks/wasb.py]: Hi @ephraimbuddy, thanks a lot for reviewing. 

Using a delimiter other than the empty string `''` might lead to non-terminal blob _paths_. Would you suggest declaring the delimiter as a local variable, and then passing it to `get_blobs_list` instead?

Comment (ephraimbuddy) [reply airflow/providers/microsoft/azure/hooks/wasb.py]: Yes @monti-python , I suggest that the delete_file method takes a delimiter argument with default of ''(empty string)

Comment (monti-python) [reply airflow/providers/microsoft/azure/hooks/wasb.py]: Brilliant, so the user could limit the deletion depth by specifying a custom `delimiter` and `ignore_if_missing = True`. Thanks for that @ephraimbuddy ! PR updated

File: tests/providers/microsoft/azure/hooks/test_wasb.py [hunk 2]
```diff
@@ -234,10 +234,11 @@ def test_delete_single_blob(self, delete_blobs, mock_service):
     @mock.patch.object(WasbHook, 'check_for_blob')
     def test_delete_multiple_blobs(self, mock_check, mock_get_blobslist, mock_delete_blobs):
         mock_check.return_value = False
+        # below return value only true if delimiter == ''
```
Comment (ephraimbuddy) [inline tests/providers/microsoft/azure/hooks/test_wasb.py]:
```suggestion
       
```

**Review by ephraimbuddy — APPROVED** [review]

**Comment by ephraimbuddy** [comment]
Congrats @monti-python :rocket:

**Pull request merged**